core/sql/generator/GenResources.cpp

```diff
@@ -83,12 +62,13 @@ static ExScratchDiskDrive * genScratchDriveList(const NAString &def,
   // end temporary
 
   const char *str = def.data();
-  if (!*str)
+  if (!str || str=="")
```

Comment:
Doesn't "str == """ mean, compare the address str to the address of the constant, empty string? If so, then this test is always false. I suspect what you intended to test is, "*str == '\0'"